### PR TITLE
New dialog for user assisted game searching

### PR DESCRIPTION
**Pull request by ghost**
New dialog that describes the game search process more informatively, informs the user about the importance of the Program Files (x86) folder, and removes the unnecessary, misleading progress bar.

### CODE REVIEW

**Comment by TracerDS** [comment]
Why do we even care about the gta being put in `Program Files (x86)`?

File: Client/loader/Utils.cpp [hunk 1]
```diff
@@ -527,30 +528,26 @@ bool LookForGtaProcess(SString& strOutPathFilename)
...
+static const SString DoUserAssistedSearch() noexcept
 {
-    SString strResult;
+    SString result;
 
-    ShowProgressDialog(g_hInstance, _("Searching for Grand Theft Auto San Andreas"), true);
+    MessageBox(nullptr, _("Start Grand Theft Auto: San Andreas.\nEnsure the game is placed in the 'Program Files (x86)' folder."), _("Searching for GTA: San Andreas"), MB_OK | MB_ICONINFORMATION);
 
-    while (!UpdateProgress(0, 100, _("Please start Grand Theft Auto San Andreas")))
+    while (true)
```
Comment (TracerDS) [inline Client/loader/Utils.cpp:537]: maybe do while instead?

**Review by TracerDS — CHANGES_REQUESTED** [review]

**Comment by MegadreamsBE** [comment]
> Why do we even care about the gta being put in `Program Files (x86)`?

It's not actually necessary. However, for most players with this issue, this is the solution. Some players install their GTA SA in OneDrive, or put it in another obscure locations.

**Comment by TracerDS** [comment]
> It's not actually necessary. However, for most players with this issue, this is the solution. Some players install their GTA SA in OneDrive, or put it in another obscure locations.

The only issue where the location might eb a problem is the onedrive folder or folders that you as the user dont have access to.
Any other location is fine. This PR should at most inform the user to put gtasa in the **local drive** folder instead of some cloud folder.

**Review by Fernando-A-Rocha — APPROVED** [review]

**Comment by Dutchman101** [comment]
Actually, im sure that in the last couple months, a change broke GTA:SA installations in non-standard install locations/drives (not talking about cloud or OneDrive, it takes much less to bug out). Users that never had any issues nor relocated their GTA install started to get the failed to find game dialog, there was a huge uptick in support requests and it wasn't even something they had in common like a specific source or version of the game.

Needs more investigation and an issue report filed. It may also be triggered by AC changes, making it harder to identify and requiring investigation from a select few people from the team. It began around the time various security fixes were added.

**Pull request merged**